firmware/arduino/chessboard/movementFunctions.ino

```diff
@@ -1,72 +1,255 @@
 bool moveDirect(int startCol, int startRow, int endCol, int endRow)
 {
-    return true;
+  // TODO:
+  // Must account for diagonal and straight movement cases
+  // Must also return true if the current position is the target position 
+  // (Since no change is needed)
+  return true;
 }
 
+// This type of move is typically decomposed into a set of 4 moves:
+// ================================================================
+// 1. Move to edge from center of square
+// 2. Move along X axis (optional)
+// 3. Move along Y axis (optional)
+// 4. Move to center of square from edge
+
+// Some of these moves may not happen, which is accounted for below
 bool moveAlongEdges(int startCol, int startRow, int endCol, int endRow)
 {
+  // There are 4 points total, but only 3 total for each x/y component
+  uint8_t rows[5], cols[5];
+  int8_t diagDirX, diagDirY;
+  int8_t deltaX, deltaY, subDeltaX, subDeltaY, absDeltaX, absDeltaY;
+  uint8_t i, statusCodeResult;
+  uint8_t curCol, curRow;
+  uint8_t numPoints = 0;
+
+  curCol = currPositionX / (stepsPerUnitSpace * 8);
+  curRow = currPositionY / (stepsPerUnitSpace * 8);
```

Comment:
It can't hurt, I'll add it.

Reply:
I'm actually not sure about this. Is it really necessary to check twice? Might it be better to check in moveAlongEdges and not in the primitives? To my knowledge, nothing should be able to make it unaligned, especially not between moveAlongEdges and a primitive call.

Reply:
On second thought, since this is checked in the primitives, it doesn't necessarily need to be done here, but it would save some computation (building the list, etc) in the event that we are misaligned. I think it should still be checked in the primitives since we might get misaligned between primitive function calls but not before ending in moveAlongEdges.

tl;dr: up to you whether you want to add a check to moveAlongEdges.

Reply:
We should leave it cuz it may get misaligned later on. Though, this'll get removed later whenever we implement the encoder stuff.